src/ert/gui/simulation/view/update.py

```diff
@@ -45,22 +47,34 @@ def __init__(self, data: DataSection, parent: Optional[QWidget] = None):
         self.setRowCount(len(data.data))
         self.setHorizontalHeaderLabels(data.header)
         self.setSelectionBehavior(QAbstractItemView.SelectionBehavior.SelectRows)
-        self.horizontalHeader().setSectionResizeMode(QHeaderView.ResizeToContents)
+        horizontal_header = self.horizontalHeader()
+        assert horizontal_header is not None
+        horizontal_header.setSectionResizeMode(QHeaderView.ResizeToContents)
```

Comment:
why save this as a variable?

Reply:
So that we can assert that it is None. Although `self.horizontlHeader()` is intented to be a property, it is called as a function so two subsequent calls could first return the horizontal header and next return `None`.